sqlglot/dialects/clickhouse.py

```diff
@@ -252,6 +252,17 @@ def _parse_primary_key(
                 wrapped_optional=wrapped_optional or in_props, in_props=in_props
             )
 
+        def _parse_on_property(self) -> t.Optional[exp.Property]:
+            if self._match_text_seq("CLUSTER"):
+                return self._parse_on_cluster()
```

Comment:
if this the on property parser doens't return anything, we should probably retreat